### DISCUSSION TITLE
Reporting Out Crystallographic Details of Alpha Titanium Variants

### DISCUSSION THREAD

**Discussion by Savillian** [ORTools4MTEX/ORTools#66] [Feature Request]
Hi everyone,

This discussion continues a previous side conversation. I wanted to ask if there was a direct way to extract the specific crystallographic plane and direction of each Alpha-Ti variant when completing variant analysis?

I'm aware the variant order may be deterministic, but I'm looking for a way to report out what the crystallographic details are for the variants in a given dataset. This way I can be sure that I am always evaluating the same variants from a given reference frame (e.g., with the 001 oriented out of the screen) and be sparred the wrath of reviewers who like precise definitions of orientations (previous experience).

I previously tried ORinfo(job.p2c) to mild success. This gave me a list of all 12 variants numerically, but only the crystallographic information for 1 of the 12 (seemingly as documentation of what crystal orientations it was permuting). The workspace object **ans.variants.orientation** also returned the variant orientations in Euler angles, which when converted gave me nonsensical plane and direction values for each variant. For example: (1 -2  1  3)  [-3 8 -5 8]

Is there an easier way to do this? I feel like I keep dancing around something obvious. The previous side conversation alluded to how the variants are ordered when a dataset has an OR close to the ideal Burgers OR. What is this order? Knowing the crystallographic identities of each variant in this ordering would be perfect and solve my dilemma. So far, my exploration of ORTools and MTEX functions for this list has turned up empty. 

As a back-up option, I was thinking of not relying on a fixed identity of each variant. Instead, I would qualitatively evaluate where each variant is in the specimen reference frame, how this is related to processing, and then look at the variant frequency for evidence of selection in response to said processing. This would also prevent semi-equivalent variants (e.g., the two sets of Alpha-Ti 45 degrees off the (001) axis of Beta-Ti) from interacting with one another statistically. In theory they should all be equivalent, minus the parent orientations they formed off and what kind of plastic slip or solute ordering they may have encountered prior to transformation.

Happy to talk further, and thanks in advance for the help!

All the best,
Alec

**Comment by frankNiessen** [comment]
Hi Alec, 

I don't have much time today to reply (Christmas season is near) but I think the reply to this post should help you:  #23 

Best wishes
Frank

**Reply by Savillian** [reply]
Hi Frank,

This looks just like what I needed! I'll give it a try today and mark this as the answer if it works!

Happy holidays!
Alec

**Reply by AzdiarGazder** [reply]
Hi @Savillian,

Since this kind of information has been previously requested by other users, the ORinfo function could be modified to return it by default.

As Frank mentioned, we're all currently signing off for the holidays. So it'll be available in the new year. 😊

Warm regards,
Azdi

**Comment by AzdiarGazder** [comment]
Hi @Savillian,

Happy new year!

I've implemented the changes as discussed previously.
Please download the latest developer version of ORTools to see the updated output from the ORinfo function.

BTW - you can also output a structure variable containing all OR information using:
```matlab
OR = ORinfo(job.p2c);
```

<ins>**Lastly, and importantly, if an explanation from the maintainers resolves your issue, please press the "Mark as answer" button on the appropriate response. Your assistance in this regard enables us to close discussions as complete. **</ins>

Hope this helps.

Warm regards,
Azdi

**Reply by Savillian** [reply]
Hi Azdi,

Happy New Year! This should be exactly what I needed

Thank you and the team for this post-holiday gift!

All the best,
Alec

**Reply by AzdiarGazder** [reply]
Hi @Savillian,

Great! 😊

Last point - could you please mark the relevant answer so that we can close this discussion thread?

Thank you for this in advance.

Warm regards,
Azdi

**Reply by Savillian** [reply]
Hi Azdi,

I closed the discussion. Since this was moved to Feature Requests I didn't see an answer option, but figured this would be the next best thing.

Thanks again!
Alec
